project/CodeStone1125/README.md

```diff
@@ -0,0 +1,166 @@
+compression. The algorithm for quadtree image compression involves producing
+a new representation of an image based on the given image, a specified 
+threshold, and the color information stored in the tree nodes. 
+The key steps of the algorithm typically include: 
+1. Start with the entire image as the root node of the quadtree.
+2. Calculate the average color (usually RGB values) for the pixels within
+    the current node.
+3. Calculate the error or difference between the pixel colors and the
+   average color within the node.
+4. Compare the error with a predefined threshold. If the error is below the
+   threshold, the node is considered a leaf node and does not get further subdivided.
+5. If the error is above the threshold, the node is subdivided into four child
+    nodes, and the process is recursively applied to each child node.
+6. Continue the process until all leaf nodes are below the error threshold,
+    effectively representing regions of the image where the colors are similar.
+
+#### Why Quadtrees for image compression
+Putting images into a tree data-structure is quite useful for many reasons. 
+To start, they allow reducing the size of an image without necessarily 
+reducing quality, as stated above in bold. This works best for images where 
+similar colors take up a large portion of space like in the top left of the
+image bellow where despite being many large subregions, they are hard to see.
+
+
+| ![image](https://github.com/CodeStone1125/nsdhw_23au/assets/72511296/b66ec4a2-0195-419f-b051-3b451d1347cc) | 
+|:-----------------------------------:|
+| **Quadtree sakura** |
+
+Trees, including quadtrees, are known for expediting processes. Quadtrees 
+specialize in focusing on the detailed sections of an image, particularly 
+beneficial for speeding up edge detection. When performing edge detection 
+on an image represented in a quadtree, we need to examine its leaf nodes and 
+their parents.
+
+Furthermore, quadtrees possess the unique ability to dynamically fine-tune 
+image compression precision, delicately balancing the level of detail with
+the consumption of computational resources. This adaptability is of paramount
+importance, especially in resource-constrained domains such as IoT and machine
+learning[1]. By elevating the threshold of the quadtree, we can craft images 
+with reduced intricacy in less critical areas, thus conserving valuable computing
+resources. Consequently, quadtrees emerge as invaluable assets in the realm of 
+image compression.
+
+### Why C++
+The rationale behind implementing this project in C++ is primarily driven by 
+the substantial computational demands.
+
+As the number of iterations increases, the growth in the number of leaf nodes
+within the Quadtree follows an exponential pattern. For each leaf node, we need
+to calculate `node.error` using the following formula:
+
+$$MSE = Σ[(Ri - R_avg)^2 + (Gi - G_avg)^2 + (Bi - B_avg)^2] / N$$
```

Comment:
(Minor issue)

This is not how we usually use the "math mode", which is basically [latex](https://www.latex-project.org).  It renders ugly and wrongly:
<img width="482" alt="image" src="https://github.com/yungyuc/nsdhw_23au/assets/399122/d2653181-f82b-40a6-b99b-5cd976bdf2f7">

A right way to write it is:

$$\mathrm{MSE} = \sum \frac{(R_i - R_{\mathrm{avg}})^2 + (G_i - G_{\mathrm{avg}})^2 + (B_i - B_{\mathrm{avg}})^2}{N}$$

When coding for numerical calcuations, lousy equations often lead to lousy results.  The project is for the software engineering of making a numerical system so I would not emphasize the mathematics.  Just to help you know more about things behind the scene.

Reply:
Sure, the issue would be solved.